xfields/ibs/_equilibrium.py

```diff
@@ -253,191 +253,207 @@ def compute_emittance_evolution(
+        if emittance_coupling_factor == 0:
+            LOGGER.warning("As 'emittance_coupling_factor` is zero, providing 'emittance_constraint' has no effect!")
+    # ---------------------------------------------------------------------------------------------
     # Handle initial transverse emittances and potential effect of coupling / excitation constraints
     # TODO: I don't like this, would rather force the user to provide gemitt_x, gemitt_y & gemitt_zeta
     if initial_emittances is None:
-        print("Emittances from the Twiss object are being used.")
+        _print("Emittances from the Twiss object are being used.")
         emittance_x, emittance_y, emittance_z = (
             twiss.eq_gemitt_x,
             twiss.eq_gemitt_y,
             twiss.eq_gemitt_zeta,
         )
-        # If emittance_coupling_factor is non zero, then natural emittance is
-        # modified accordingly
-        # fmt: off
-        if emittance_coupling_factor != 0 and emittance_constraint.lower() == "coupling":
-            # The convention used is valid for arbitrary damping partition numbers and emittance_coupling_factor.
+        # If emittance_coupling_factor is non zero, then natural emittance is modified accordingly (used
+        # convention is valid for arbitrary damping partition numbers and emittance_coupling_factor).
+        if emittance_constraint.lower() == "coupling" and emittance_coupling_factor != 0:
             emittance_y = emittance_x * emittance_coupling_factor / (1 + emittance_coupling_factor * twiss.partition_numbers[1] / twiss.partition_numbers[0])
             emittance_x *= 1 / (1 + emittance_coupling_factor * twiss.partition_numbers[1] / twiss.partition_numbers[0])
 
-        if emittance_coupling_factor != 0 and emittance_constraint.lower() == "excitation":
+        if emittance_constraint.lower() == "excitation" and emittance_coupling_factor != 0:
             # The convention used only enforce a constraint on the vertical
             # emittance
             emittance_y = emittance_x * emittance_coupling_factor
     else:
         emittance_x, emittance_y, emittance_z = initial_emittances
-        # fmt: on
-    # ----------------------------------------------------------------------------------------------
+    # fmt: on
+    # ---------------------------------------------------------------------------------------------
     # Handle initial longitudinal emittance and potential effect of bunch lengthening
     sigma_zeta = (emittance_z * twiss.bets0) ** 0.5
     sigma_delta = (emittance_z / twiss.bets0) ** 0.5
     if overwrite_sigma_zeta is not None:
-        warnings.warn("'overwrite_sigma_zeta' is specified, make sure it remains consistent with 'initial_emittances'.")
+        LOGGER.warning("'overwrite_sigma_zeta' is specified, make sure it remains consistent with 'initial_emittances'.")
```

Comment:
A log of level WARNING is emitted by default so this is not hiding information. Only reason it wouldn't show is if the user specifically sets up to hide these logs, in which case not our problem